metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -94,6 +106,85 @@ public void addEdge(@Nonnull final Edge edge) {
     executeStatements(statements);
   }
 
+  @Nonnull
+  @Override
+  public EntityLineageResult getLineage(@Nonnull Urn entityUrn, @Nonnull LineageDirection direction,
+                                        GraphFilters graphFilters, int offset, int count, int maxHops) {
+    log.debug(String.format("Neo4j getLineage maxHops = %d", maxHops));
+
+    final String statement = generateLineageStatement(entityUrn, direction, graphFilters, maxHops);
+
+    List<Record> neo4jResult = statement != null ? runQuery(buildStatement(statement, new HashMap<>())).list() : new ArrayList<>();
+
+    neo4jResult = neo4jResult.stream()
+            .collect(Collectors.toMap(item -> item.values().get(2).asNode().get("urn").asString(),
```

Comment:
Is the "get(2)" safe here? Expecting the item.values array to _always_ have more than one item?

Reply:
Yes. The query returns array of 3 elements: start node, end node and relation.